Refactor/movie details

baraa0abd opened the pull request:
Fixes

- Fixed review date formatting
- Resolved reviewer image placeholder issues
- when there is no review it shouldn't shows "there is no review yet" , now it will hide the whole section from the user
- fix the color of chip cards (that represent the rating , date ..)
- refactor: format review date and remove unused code
- separate rating code from the movie details to another files (presentation/src/main/java/com/madrid/presentation/component/rating/ratingBottomsheetContent.kt   &   presentation/src/main/java/com/madrid/presentation/component/rating/ratingSuccessBottomsheet.kt)

baraa0abd committed:
fix: Adjust MoviePosterDetailScreen padding and image size

baraa0abd committed:
feat: Implement Rating Bottom Sheet with Success State

This commit introduces a new Rating Bottom Sheet feature.

- Adds `RatingBottomSheetContent` composable to handle both guest and authenticated user rating flows.
- Adds `RatingSuccessBottomSheetContent` composable to display a success message after rating submission.
- Integrates the new bottom sheet into `MovieDetailScreen`.
- Refactors `MovieDetailsHeader` for minor layout adjustments.

baraa0abd committed:
Refactor: Relocate files and update imports

- Relocated ratingSuccessBottomsheet.kt and ratingBottomsheetContent.kt from `screens.rating` to `component.rating`.
- Relocated MovieDetailScreen.kt from `screens.detailsScreen.detailsMovieScreen` to `screens.detailsScreen.movieDetail` and renamed it to MovieDetail.kt.
- Relocated logout.kt from `screens.logOut` to `component.logout`.
- Updated import statements in MovieDetail.kt, MovioNavHost.kt, and MoreScreen.kt to reflect the new file locations.
- Replaced `Alignment.CenterHorizontally` with `CenterHorizontally` in ratingSuccessBottomsheet.kt and ratingBottomsheetContent.kt.

baraa0abd committed:
refactor: format review date and remove unused code

The `formatReviewDate` function was moved from the presentation layer to the data layer.
The "Read More" and "Read Less" functionality was removed from the `ReviewCard` composable.
Removed unused imports and logs.

baraa0abd committed:
fix: Movie review date format

The date format for movie reviews was inconsistent.
This change ensures that all movie review dates are formatted as "MMM dd, yyyy".
It also updates the `reviewerPhotoUrl` in the `Review` entity to be nullable to handle cases where the URL might be missing.

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

# Conflicts:
#	data/src/main/java/com/madrid/data/repositories/MovieRepositoryImpl.kt

baraa0abd committed:
fix: update placeholder image and remove unused imports

The commit message summarizes the changes in the diffs:
- The placeholder image is updated in `FilteredImage.kt`, `ReviewCard.kt`, and `ReviewsScreenUiState.kt`.
- Unused imports are removed from `RateMapper.kt` and `MovieRepositoryImpl.kt`.

baraa0abd committed:
fix: display placeholder for null reviewer image URLs and enhance review card UI

- Show a placeholder image if the reviewer's image URL is null in `ReviewScreen` and `ReviewsScreen`.
- In `ReviewCard`:
    - Ensure `reviewerImageUrl` is not blank before attempting to load it.
    - Set content descriptions for reviewer image and rating icon.
    - Truncate long reviewer names with an ellipsis.
    - Implement "Read more"/"Read less" functionality for review content.

baraa0abd committed:
Refactor: Rename ReviewScreen to ReviewSection and enhance ReviewCard UI

- Renamed `ReviewScreen.kt` to `ReviewSection.kt`.
- In `ReviewSection.kt` and `ReviewsScreen.kt`:
    - Pass `reviewerImageUrl` directly to `ReviewCard` without placeholder fallback.
- In `ReviewCard.kt`:
    - `reviewerImageUrl` parameter is now nullable.
    - Load reviewer image only if `reviewerImageUrl` is not null or blank.
    - Format rating to one decimal place.
    - Adjusted text color for rating and review content.
    - Removed "Read more"/"Read less" toggle for review content and directly display content with potential ellipsis based on `isExpanded` state (though the `isExpanded` logic seems to have been removed).

baraa0abd committed:
Refactor ReviewCard and ReviewsScreen

- Updated ReviewCard to handle null or empty reviewerImageUrl by displaying a default user icon.
- Removed "Read More" functionality and maxLines restriction from review content in ReviewCard.
- Adjusted spacing and padding in ReviewCard for better visual layout.
- Refactored ReviewsScreen to display a message when no reviews are available.
- Moved ReviewScreen composable from `reviewsScreen` package to `reviewsScreen.composables`.
- Added a new drawable resource `ic_no_user.xml` for the default user icon.
- Updated `ReviewUiState` to make `reviewerImageUrl` nullable.

baraa0abd committed:
Refactor ReviewCard and ReviewsScreen

- Updated ReviewCard to handle null or empty reviewerImageUrl by displaying a default user icon.
- Removed "Read More" functionality and maxLines restriction from review content in ReviewCard.
- Adjusted spacing and padding in ReviewCard for better visual layout.
- Refactored ReviewsScreen to display a message when no reviews are available.
- Moved ReviewScreen composable from `reviewsScreen` package to `reviewsScreen.composables`.
- Added a new drawable resource `ic_no_user.xml` for the default user icon.
- Updated `ReviewUiState` to make `reviewerImageUrl` nullable.

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into refactor/movie-details

baraa0abd committed:
Refactor ReviewScreen and ReviewSection

- Renamed `ReviewsScreen.kt` to `ReviewsSection.kt` and updated its package.
- Renamed `ReviewSection.kt` to `ReviewScreen.kt` and updated its package.
- Updated imports in `MovieDetail.kt`, `SeriesDetailsScreen.kt`, and `MovioNavHost.kt` to reflect the name changes.
- Conditionally render review sections and empty messages based on whether reviews are available.
- Used `MovioArtistsCard` in `ReviewCard.kt` for displaying reviewer avatars.

baraa0abd committed:
Refactor: Move ReviewScreen and ReviewsSection to correct packages

The `ReviewScreen` composable was moved from the `composables` subpackage to its parent `reviewsScreen` package.
Conversely, the `ReviewsSection` composable was moved from the `reviewsScreen` package to its `composables` subpackage.
Import statements across the affected files were updated to reflect these changes.

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

fatimafaisalghazi commented on detectImageContent/src/main/java/com/madrid/detectImageContent/FilteredImage.kt:
```diff
@@ -39,7 +39,7 @@ fun FilteredImage(
     val detection = remember(context) { SensitiveContentDetection(context) }
 
-    if (imageUrl.isNotBlank()) {
+    if (imageUrl?.isNotBlank() == true) {
```
i think if u make it  if (imageUrl.isNotBlank()) much better

baraa0abd replied:
it's already there , plz update the branch

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/reviewsScreen/composables/ReviewCard.kt:
```diff
@@ -96,43 +109,33 @@ fun ReviewCard(
+                textStyle = Theme.textStyle.label.smallRegular12,
                 modifier = Modifier
-                    .clickable { isExpanded = !isExpanded }
                     .padding(top = 4.dp)
```
i dont understand why do you remove  text = if (isExpanded) stringResource(R.string.read_less) else stringResource(R.string.read_more), 
in the figma design it must have this 

the textStyle  and color was right ,pleas return it to what it was

<img width="678" height="696" alt="image" src="https://github.com/user-attachments/assets/af9299c9-46c0-4438-96e9-264b1a18d4d5" />

baraa0abd replied:
it's already there , plz update the branch

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/reviewsScreen/composables/ReviewCard.kt:
```diff
@@ -96,43 +109,33 @@ fun ReviewCard(
 
         Spacer(modifier = Modifier.height(12.dp))
 
-        MovioText(
-            text = content,
-            color = Theme.color.surfaces.onSurfaceVariant,
-            textStyle = Theme.textStyle.label.smallRegular12,
-            maxLines = maxLines,
-            modifier = Modifier.fillMaxWidth()
-        )
-
-        if (showReadMore) {
-            Spacer(modifier = Modifier.height(4.dp))
```
this is for the content and it like the design

baraa0abd replied:
it's already there , plz update the branch

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/utils/RateFormatter.kt:
```diff
@@ -1,6 +1,10 @@
 package com.madrid.presentation.utils
 
+import android.icu.text.SimpleDateFormat
+import kotlinx.datetime.LocalDateTime
 import java.text.DecimalFormat
+import java.time.format.DateTimeFormatter
+import java.util.Locale
 import kotlin.math.floor
```
i dont understand why we use 
import java.time.format.DateTimeFormatter
and
import kotlinx.datetime.LocalDateTime

baraa0abd replied:
plz update the branch

fatimafaisalghazi commented:
pleas add descrpition and link this to an issue

baraa0abd committed:
Refactor FilteredImage composable

- Made imageUrl parameter non-nullable.
- Removed default placeholder image.

baraa0abd committed:
Remove `ic_no_user.xml`

fatimafaisalghazi commented:
`delete read less or read more from the review screen (based on figma)
`

The "more / less' is already available in the Figma design.

baraa0abd committed:
feat: add "read more" functionality to review card

This commit introduces a "read more/read less" feature to the `ReviewCard` composable.

- If the review content exceeds 100 characters, a "Read more" text is displayed.
- Clicking "Read more" expands the content to show the full review and changes the text to "Read less".
- Clicking "Read less" collapses the content back to the initial truncated view.

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into refactor/movie-details

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into refactor/movie-details

# Conflicts:
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/seriesDetails/SeriesDetailsScreen.kt

baraa0abd committed:
Refactor: Remove unused rate and date formatting functions

The `formatRate(String)` and `formatReviewDate(String)` functions in `RateFormatter.kt` were removed as they were no longer in use.

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

# Conflicts:
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/movieDetail/MovieDetail.kt
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/seriesDetails/SeriesDetailsScreen.kt

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/reviewsScreen/composables/ReviewCard.kt:
```diff
@@ -65,13 +68,31 @@ fun ReviewCard(
-                    .clip(CircleShape)
-                    .size(32.dp)
-            )
+            if (reviewerImageUrl?.isNotBlank() == true) {
```
i think we can remove  == true

baraa0abd replied:
it will shows an error

baraa0abd replied:
<img width="941" height="193" alt="image" src="https://github.com/user-attachments/assets/504e187c-5d23-4a00-87b3-331926405fd5" />

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/reviewsScreen/composables/ReviewCard.kt:
```diff
@@ -96,43 +119,44 @@ fun ReviewCard(
                 MovioIcon(
                     painter = painterResource(id = R.drawable.bold_star),
-                    contentDescription = null,
+                    contentDescription = "Rating",
```
use string resorce

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/reviewsScreen/composables/ReviewCard.kt:
```diff
@@ -65,13 +68,31 @@ fun ReviewCard(
+                        modifier = Modifier
+                            .clip(CircleShape),
+                        circleImageSize = 118.dp,
+                        artistsName = reviewerName,
```
i dont undeerstand why do we use MovioArtistsCard here and i noticed that  artistsName is  reviewerName,
and it show always the place holder

baraa0abd replied:
which component i should use unstead?

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/reviewsScreen/composables/ReviewCard.kt:
```diff
@@ -96,43 +119,44 @@ fun ReviewCard(
+            if (showReadMore) {
+                Spacer(modifier = Modifier.height(4.dp))
+                MovioText(
+                    text = if (isExpanded) stringResource(R.string.read_less) else stringResource(R.string.read_more),
```
in the design its less and more 
without read

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/movieDetail/MovieDetail.kt:
```diff
@@ -107,7 +93,6 @@ fun MovieDetailsScreen(
     var showAddRatingBottomSheet by remember { mutableStateOf(false) }
-    var showDoneRatingBottomSheet by remember { mutableStateOf(false) }
 
     if (uiState.isLoading) {
```
if it is loading use DialogWithButtonLayout instead of EmptySearchLayout

baraa0abd replied:
didn’t get it 😅

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

baraa0abd committed:
Fix: remove extra padding in `ReviewCard`

This commit fixes an issue where an extra padding was added
between the image and the text in the `ReviewCard` composable.
This was caused by a wrong value being passed to the
`paddingBetweenImageAndText` parameter.

The `paddingBetweenImageAndText` parameter in the `ReviewCard`
composable is now set to 0.dp to remove the extra padding.

Additionally, this commit updates the import path for
`MovieDetailsScreen` in `MovioNavHost.kt`.

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

baraa0abd committed:
mv rating  formatting from the mapper in data to utils that used in view model and uistate to represent the date formatter

baraa0abd committed:
fix date formate in review screen

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

# Conflicts:
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/reviewsScreen/composables/ReviewCard.kt

baraa0abd committed:
fix img place holder

baraa0abd committed:
make reviewerImageUrl un nullable

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

baraa0abd committed:
refactor: localize success message and update addtolist screen

- Localized success message in `SuccessNotificationRow` and added translations for Arabic.
- Refactored `addToList.kt` to remove commented out code and improve LaunchedEffect logic.
- Updated `BottomMediaActions` to use `SpaceBetween` arrangement and adjust play icon size.
- Removed default fake data and preview from `ListSelectionContent`.

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

# Conflicts:
#	data/src/main/java/com/madrid/data/repositories/MovieRepositoryImpl.kt

baraa0abd committed:
fix: handle null or empty time and rate in MovieDetailsHeader

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

# Conflicts:
#	designSystem/src/main/res/values-ar/strings.xml
#	designSystem/src/main/res/values/strings.xml
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/movieDetail/MovieDetail.kt

baraa0abd committed:
fix: navigate to login screen from guest rating

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

yousef-osama11 commented on presentation/src/main/java/com/madrid/presentation/viewModel/detailsViewModel/DetailsMovieViewModel.kt:
```diff
@@ -149,20 +148,17 @@ class DetailsMovieViewModel @Inject constructor(
-                        date = review.date,
-                        content = review.comment,
-                    )
+                // Option 1: Using the extension function (Recommended)
```
remove unnecessary comments

baraa0abd committed:
refactor: remove comment in DetailsMovieViewModel

baraa0abd committed:
Merge branch 'develop' into refactor/movie-details

baraa0abd committed:
Merge remote-tracking branch 'origin/refactor/movie-details' into refactor/movie-details

yousef-osama11 reviewed (approved):
عاشت ايدج

baraa0abd committed:
Refactor: Remove rating BottomSheet and update success notification message

MostafaMohamed2002 committed:
Merge branch 'develop' into refactor/movie-details

MostafaMohamed2002 reviewed (approved):
LGTM!
